node/tests/utils.rs

```diff
@@ -478,45 +478,43 @@ impl MASQNode {
         args
     }
 
-    fn virtual_arg_pairs(args: Vec<String>) -> Vec<(String, Option<String>)> {
-        fn look_at_val_distanced_by_two<I>(
-            peekable: &mut Peekable<I>,
-            current_string: String,
-            mut acc: Vec<(String, Option<String>)>,
-        ) -> Vec<(String, Option<String>)>
-        where
-            I: Iterator<Item = String>,
-        {
-            match peekable.peek() {
-                //possibly value for that arg
-                Some(next_string) => {
-                    if next_string.starts_with("--") {
-                        acc.push((current_string, None));
-                        acc
-                    } else {
-                        let value = peekable.next().unwrap();
-                        acc.push((current_string, Some(value)));
-                        acc
+    fn virtual_arg_pairs_with_remembered_position(
+        args: Vec<String>,
+    ) -> Vec<(String, PositionedArg)> {
+        let shifted_by_one = args
+            .clone()
+            .into_iter()
+            .map(|val| Some(val))
+            .skip(1)
+            .chain(once(None));
```

Comment:
Comment about how this chain is here so that a parameter without a value at the end will be obvious.